src/Rector/Class_/PreferPHPUnitThisCallRector.php

```diff
@@ -73,8 +75,10 @@ public function refactor(Node $node): ?Node
         }
 
         $hasChanged = false;
-        $this->traverseNodesWithCallable($node, function (Node $node) use (&$hasChanged): ?MethodCall {
-            if (! $node instanceof StaticCall) {
+        $isStatic = false;
+        $this->traverseNodesWithCallable($node, function (Node $node) use (&$hasChanged, &$isStatic): ?MethodCall {
+            $isStatic = ($isStatic) || ($node instanceof ClassMethod && $node->isStatic()) || ($node instanceof Closure && $node->static);
```

Comment:
oh, `$isStatic` must always be resetted on next visit next method, the ($isStatic) first condition must be removed. The NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN should actually can be used when visit ClassMethod or Closure with isStatic condition = true. Could you create another fixture with multiple test, one with static, another one without static method?

Reply:
What kind of extra test fixture would you like to see we already have a replace fixture

Reply:
should be something like this:

```php
    public static function testMe()
    {
        self::assertSame(1, 1);
    }

    public function testMe2()
    {
        self::assertSame(1, 1);
    }
```

The `testMe2` should use `$this->assertSame()`